src/cli/self_update.rs

```diff
@@ -1103,6 +1106,32 @@ pub fn get_available_rustup_version() -> Result<String> {
     Ok(String::from(available_version))
 }
 
+pub fn check_rustup_update() -> Result<()> {
```

Comment:
Just an idea, what about putting this in self-update:

pseudocode:

```
fn self_update(mode: UpdateMode) -> Result<()> {
   if #[cfg(no_self_update)] {
       return Ok(());
   }  else {
      match mode {
         UpdateMode::Disable => { return Ok(()) },
         UpdateMode::CheckOnly => { ... check logic ... }
         UpdateMode::Enable => { ... update logic ... {
     }
  }
}
```

Then you don't need a separate call for checking self updates and can have all the logic about whether self update are possible or not in just one place.

Reply:
This fn is also used in `rustup check`, so I think we can keep it.

Reply:
The check in the rustup check does not actually check UpdateMode. So I don't think it's necessary to couple it?

Reply:
The logic you describe is actually only used in the update method. The value of UpdateMode is not considered elsewhere.
https://github.com/rust-lang/rustup/blob/330a5b2a2bb60389747e58e82c4646bc0a5bd13d/src/cli/rustup_mode.rs#L933